source/unix/openssl_platform_init.c

```diff
@@ -48,6 +69,25 @@ extern int HMAC_Update(HMAC_CTX *, const unsigned char *, size_t) __attribute__(
 extern int HMAC_Final(HMAC_CTX *, unsigned char *, unsigned int *) __attribute__((weak, used));
 extern int HMAC_Init_ex(HMAC_CTX *, const void *, int, const EVP_MD *, ENGINE *) __attribute__((weak, used));
 
+static void s_hmac_ctx_reset_openssl(HMAC_CTX *ctx) {
+    AWS_PRECONDITION(ctx);
+
+    int (*reset_ptr)(HMAC_CTX *) = (int (*)(HMAC_CTX *))g_aws_openssl_hmac_ctx_table->impl.reset_fn;
+    reset_ptr(ctx);
+}
+
+static int s_hmac_init_ex_openssl(HMAC_CTX *ctx, const void *key, size_t key_len, const EVP_MD *md, ENGINE *impl) {
+    AWS_PRECONDITION(ctx);
+
+    int (*init_ex_ptr)(HMAC_CTX *, const void *, int, const EVP_MD *, ENGINE *) =
```

Comment:
add comment like 
`/* openssl takes key_len as an int, rather than size_t */`

also, need to defend against `key_len > INT_MAX`

Reply:
Yikes, looks like the only place that calls this function naively casts it to an int, we should change that

https://github.com/awslabs/aws-c-cal/blob/55e478b31cf50517cd359c1ef2e79ef6c2a1f9d0/source/unix/opensslcrypto_hmac.c#L76